### ISSUE TITLE
Watcom Additional Warnings - Sqash'em or Heed'em?

### ISSUE THREAD

**Issue by arstrube** [cpputest/cpputest#765]
```c++
src\CppUTest\SimpleString.cpp
 (109): Warning! W391: col(19) assignment found in boolean expression
 (209): Warning! W391: col(25) assignment found in boolean expression
 ------------------------------
tests\PluginTest.cpp
 (133): Warning! W368: col(63) conditional expression in if statement is always false
 (133): Warning! W013: col(63) unreachable code
 ... (many more of these in different test cases)
```
@basvodde what do you think?

**Comment by arstrube** [comment]
I vote to squash them, especially W368/W013...

**Comment by arstrube** [comment]
Hmm. Reconsidering.... It is apparently taking exception to this:
```c++
#define CHECK_EQUAL_LOCATION(expected, actual, text, file, line)\
  { if ((expected) != (actual)) { \
      if ((actual) != (actual)) \
      	  UtestShell::getCurrent()->print("WARNING:\n\tThe \"Actual Parameter\" parameter is evaluated multiple times resulting in different values.\n\tThus the value in the error message is probably incorrect.", file, line); \
      if ((expected) != (expected)) \
      	  UtestShell::getCurrent()->print("WARNING:\n\tThe \"Expected Parameter\" parameter is evaluated multiple times resulting in different values.\n\tThus the value in the error message is probably incorrect.", file, line); \
      UtestShell::getCurrent()->assertEquals(true, StringFrom(expected).asCharString(), StringFrom(actual).asCharString(), text, file, line); \
  } \
  else \
  { \
    UtestShell::getCurrent()->assertLongsEqual((long)0, (long)0, NULL, file, line); \
  } }
```
Although the intention is clear, this is funky code indeed ;-). There  is a test for this, so we know it works. However, I am as puzzled as the poor compiler. Where is the hidden side-effect?

**Comment by arstrube** [comment]
Hmm. `actual` in this case is `_countingMethod()` so the compiler can't possibly know that it is the same. Will need to squash'em after all.....

**Comment by arstrube** [comment]
Fixed by #766
